command/pr_lookup.go

```diff
@@ -0,0 +1,94 @@
+package command
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/context"
+	"github.com/cli/cli/git"
+	"github.com/cli/cli/internal/ghrepo"
+)
+
+func prFromArgs(ctx context.Context, apiClient *api.Client, repo ghrepo.Interface, args []string) (*api.PullRequest, error) {
+	if len(args) == 0 {
+		return prForCurrentBranch(ctx, apiClient, repo)
+	}
+
+	// First check to see if the prString is a url
+	prString := args[0]
+	pr, err := prFromURL(ctx, apiClient, repo, prString)
+	if pr != nil || err != nil {
+		return pr, err
+	}
+
+	// Next see if the prString is a number and use that to look up the url
+	pr, err = prFromNumberString(ctx, apiClient, repo, prString)
+	if pr != nil || err != nil {
+		return pr, err
+	}
+
+	// Last see if it is a branch name
+	return api.PullRequestForBranch(apiClient, repo, "", prString)
+}
+
+func prFromNumberString(ctx context.Context, apiClient *api.Client, repo ghrepo.Interface, s string) (*api.PullRequest, error) {
+	if prNumber, err := strconv.Atoi(strings.TrimPrefix(s, "#")); err == nil {
+		return api.PullRequestByNumber(apiClient, repo, prNumber)
+	}
+
+	return nil, nil
+}
+
+func prFromURL(ctx context.Context, apiClient *api.Client, repo ghrepo.Interface, s string) (*api.PullRequest, error) {
+	r := regexp.MustCompile(`^https://github\.com/([^/]+)/([^/]+)/pull/(\d+)`)
+	if m := r.FindStringSubmatch(s); m != nil {
+		prNumberString := m[3]
```

Comment:
It looks like this ignores the information about the base repo from the URL. To address issues such as https://github.com/cli/cli/issues/1060, would we consider respecting it?